lib/Doctrine/ORM/Query/Lexer.php

```diff
@@ -175,7 +178,13 @@ protected function getType(&$value)
                     }
                 }
 
-                return self::T_IDENTIFIER;
+                if (strpos($value, '\\') !== false) {
+                    return self::T_QUALIFIED_NAME;
+                } else if (strpos($value, ':') !== false) {
```

Comment:
Then it is ok that way: If there is an `\` in it, it is *not* an alias (will lead to unresolvable class in the "Alias:Sub\Class" case). Aliases only work for "Foo:Bar" like names.